pyvrp/read.py

```diff
@@ -85,29 +88,37 @@ def read(
 
     depots = instance.get("depot", np.array([0]))
     num_vehicles = instance.get("vehicles", num_clients - 1)
-    capacity = instance.get("capacity", _INT_MAX)
     edge_weight = round_func(instance["edge_weight"])
 
     if "demand" in instance:
         demands = instance["demand"]
     else:
         demands = np.zeros(num_clients, dtype=int)
 
+    capacity = instance.get("capacity", demands.sum())
+
     if "node_coord" in instance:
         coords = round_func(instance["node_coord"])
     else:
         coords = np.zeros((num_clients, 2), dtype=int)
 
-    if "time_window" in instance:
-        time_windows = round_func(instance["time_window"])
-    else:
-        time_windows = np.repeat([[0, _INT_MAX]], num_clients, axis=0)
-
     if "service_time" in instance:
         service_times = round_func(instance["service_time"])
     else:
         service_times = np.zeros(num_clients, dtype=int)
 
+    if "time_window" in instance:
+        time_windows = round_func(instance["time_window"])
+    else:
+        # Upper bound based on the maximum route duration
+        max_route_len = capacity // demands[demands > 0].min()
```

Comment:
Not sure, but there may be instances with some very small demands, how about we cap `max_route_len` at `num_clients`? Maybe also support 0 demand (i.e. TSP) and check for overflow of these values?

Reply:
I have removed the bounds from this PR but I'll add your comments to my other branch for this.